File: node/sftp/ssh2-sftp-client/sftp.js
import { parse } from "url";
import { readFile } from "fs";

var proxy = parse(process.env.QUOTAGUARDSTATIC_URL);
var puser = proxy.auth.split(':')[0];
var ppass = proxy.auth.split(':')[1];

import { SocksClient } from 'socks';
import SFTPClient from 'ssh2-sftp-client';

const host = 'test.rebex.net';
const port = 22; // default SSH/SFTP port on remote server

// connect to SOCKS 5 proxy
const { socket } = await SocksClient.createConnection({
  proxy: {
    host: proxy.hostname,
    port: 1080, // proxy port
    userId: puser,
    password: ppass,
    type: 5, // for SOCKS v5
  },
  command: 'connect',
  destination: { host, port } // the remote SFTP server
});

const client = new SFTPClient();
client.connect({
  host,
  sock: socket, // pass the socket to proxy here (see ssh2 doc)
  username: 'demo',
  password: 'password'
}).then(() => {
  console.log("Connected, downloading readme.txt");
  return client.get("readme.txt", "readme.txt");
})
.then(() => {
  console.log("File Downloaded!");
  client.end();
  readFile("readme.txt", function(err, data){
    err ? Function("error","throw error")(err) : console.log(data.toString('utf8'));
  });
})
.catch(err => {
  console.error(err.message);
});

File: ruby/sftp/sftp.rb
#!/usr/bin/env ruby

require 'net/ssh/proxy/http'
require 'net/sftp'
require 'uri'

quotaguard = URI.parse(ENV["QUOTAGUARDSTATIC_URL"])
proxy = Net::SSH::Proxy::HTTP.new(quotaguard.host, quotaguard.port, user: quotaguard.user, password: quotaguard.password)

Net::SFTP.start('test.rebex.net', 'demo', password: 'password', proxy: proxy) do |sftp|
  sftp.download!('readme.txt', 'readme.txt')
end

if File.exists?('readme.txt')
  puts "Downloaded 'readme.txt' successfully"
  puts "   CONTENTS"
  puts "--------------"
  puts File.read('readme.txt')
  puts "--------------"
else
  puts "MISSING 'readme.txt'"
  exit 1
end
